### ISSUE TITLE
Cannot add .js,.sh files to the list

### ISSUE THREAD

**Issue by spirodonfl** [ThePrimeagen/harpoon#388]
My setup

* WSL2 (Ubuntu distro)
* Fish shell
* Neovim w/ Lazy plugin manager
* Kickstart.nvim

Files that have worked for me (so far)
* md
* zig
* json
* html
* css

Files that have *not* worked for me (so far)
* js
* sh

I have not done a thorough review of various file types. The above are all the ones I've tried.

**Comment by daltondiaz** [comment]
Hi @spirodonfl, 

I tested and worked for me, both the .js and .sh.

So make sure that you're in the latest version of harpoon 2, open Neovim and execute `:lua require("harpoon").logger:show()`, generate the error again, copy and post here.

**Comment by mwishoff** [comment]
I tested as well, and wasn't able to reproduce.

**Comment by spirodonfl** [comment]
Out of curiosity, I opened up the harpoon menu, literally added the file `eventbus.js` as a new line to the list and then did `:w` to save the list... *that* worked. I can see the file in the list, I can swap to it (and from it to other files in the list) and it works even if I quit nvim and go back.

**Comment by ThePrimeagen** [comment]
@spirodonfl is this a real issue or are you just messing with me?

if it is a real issue, could you create the exact steps + log for this.  it just seems so wild! (i use it on both js and sh projects at work)

**Comment by spirodonfl** [comment]
No troll. Quite legit. I actually have video evidence.

https://www.youtube.com/watch?v=3aV_wzzTSRo&ab_channel=SpiroFloropoulos - start at 10:00 minute mark approximately

I just tried again to make sure.

I'm gonna be honest, I have a feeling this is an issue with my own local setup although I'm having a hard time seeing *how*. All I have is kickstart.nvim + harpoon. I think maybe I've somehow got my local into a weird broken state and I can't get out of it.

My logs from having tested it manually are here: https://github.com/ThePrimeagen/harpoon/issues/388#issuecomment-1844148671

You can see when I open up test.zig, the log shows it. When I open up a .js file, it doesn't even show up in the logs.

I should consider the idea that this isn't a harpoon issue but something else going on that's just *affecting* harpoon.

**Comment by daltondiaz** [comment]
> @ThePrimeagen No troll. Quite legit. I actually have video evidence.
> 
> https://www.youtube.com/watch?v=3aV_wzzTSRo&ab_channel=SpiroFloropoulos - start at 10:00 minute mark approximately
> 
> I just tried again to make sure.
> 
> I'm gonna be honest, I have a feeling this is an issue with my own local setup although I'm having a hard time seeing _how_. All I have is kickstart.nvim + harpoon. I think maybe I've somehow got my local into a weird broken state and I can't get out of it.
> 
> My logs from having tested it manually are here: [#388 (comment)](https://github.com/ThePrimeagen/harpoon/issues/388#issuecomment-1844148671)
> 
> You can see when I open up test.zig, the log shows it. When I open up a .js file, it doesn't even show up in the logs.
> 
> I should consider the idea that this isn't a harpoon issue but something else going on that's just _affecting_ harpoon.

Start here the steps to reproduce the problem: https://www.youtube.com/live/3aV_wzzTSRo?si=zOr0a2v0aDMbhlhh&t=654 btw, nice degen mustache!! 

You can test if this happens in different projects with .js and .sh files? It's possible to share your configuration with us?

**Comment by spirodonfl** [comment]
Thanks @daltondiaz 

It happens on any project, yeah.

[init.lua.zip](https://github.com/ThePrimeagen/harpoon/files/13629698/init.lua.zip)

**Comment by squ1d123** [comment]
I have exactly the same issue, basically the same setup as described.

- WSL2 (Ubuntu distro)
- zsh shell
- Neovim w/ Lazy plugin manager
- Kickstart.nvim

**Comment by spirodonfl** [comment]
@squ1d123 ok then I'll go down the rabbit hole and fix this on behalf of @ThePrimeagen. I'm betting this is a WSL issue, not a Harpoon issue, but I'll update the ticket once I've confirmed.

Might even do it on a live stream or something.

**Comment by spirodonfl** [comment]
@ThePrimeagen @squ1d123 I solved it.

Here's the issue - kickstart install nvim-treesitter.

Here is part of the config for it inside of `init.lua`
```
...
      swap = {
        enable = true,
        swap_next = {
          ['<leader>a'] = '@parameter.inner',
        },
        swap_previous = {
          ['<leader>A'] = '@parameter.inner',
        },
      },
...
```

Notice anything? Yeah. <leader>a

I updated my config from
```
vim.keymap.set('n', '<leader>a', function() harpoon:list():append() end)
```
to
```
vim.keymap.set('n', ',a', function() harpoon:list():append() end)
```

Why does this cause issues ONLY on some files? It literally just depends on how the particular filetype is configured within treesitter so *some* files don't trigger this "swap -> @parameter.inner" scenario and some do.

Just avoid using this keymap/binding and you're good to go.

@ThePrimeagen if you're happy with this, I'm good to accept this as closed/complete.
